tests/integration/golang/mlflow/namespace/flows/experiment_test.go

```diff
@@ -0,0 +1,592 @@
+					},
+				},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageActive),
+			},
+		},
+	)
+	s.getExperimentByIDAndCompare(
+		namespace2Code,
+		experiment2ID,
+		&response.GetExperimentResponse{
+			Experiment: &response.ExperimentPartialResponse{
+				ID:   experiment2ID,
+				Name: "UpdatedExperiment2",
+				Tags: []response.ExperimentTagPartialResponse{
+					{
+						Key:   "KeyTag2",
+						Value: "ValueTag2",
+					},
+				},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageActive),
+			},
+		},
+	)
+
+	// test `POST /experiments/delete` endpoint.
+	s.deleteExperiment(namespace1Code, experiment1.Experiment.ID)
+	s.deleteExperiment(namespace2Code, experiment2.Experiment.ID)
+
+	// check that experiment lifecycle has been updated.
+	s.getExperimentByIDAndCompare(
+		namespace1Code,
+		experiment1ID,
+		&response.GetExperimentResponse{
+			Experiment: &response.ExperimentPartialResponse{
+				ID:   experiment1ID,
+				Name: "UpdatedExperiment1",
+				Tags: []response.ExperimentTagPartialResponse{
+					{
+						Key:   "KeyTag1",
+						Value: "ValueTag1",
+					},
+				},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageDeleted),
+			},
+		},
+	)
+	s.getExperimentByIDAndCompare(
+		namespace2Code,
+		experiment2ID,
+		&response.GetExperimentResponse{
+			Experiment: &response.ExperimentPartialResponse{
+				ID:   experiment2ID,
+				Name: "UpdatedExperiment2",
+				Tags: []response.ExperimentTagPartialResponse{
+					{
+						Key:   "KeyTag2",
+						Value: "ValueTag2",
+					},
+				},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageDeleted),
+			},
+		},
+	)
+
+	// test `POST /experiments/restore` endpoint.
+	s.restoreExperiment(namespace1Code, experiment1ID)
+	s.restoreExperiment(namespace2Code, experiment2ID)
+
+	// check that experiment lifecycle has been updated.
+	s.getExperimentByIDAndCompare(
+		namespace1Code,
+		experiment1ID,
+		&response.GetExperimentResponse{
+			Experiment: &response.ExperimentPartialResponse{
+				ID:   experiment1ID,
+				Name: "UpdatedExperiment1",
+				Tags: []response.ExperimentTagPartialResponse{
+					{
+						Key:   "KeyTag1",
+						Value: "ValueTag1",
+					},
+				},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageActive),
+			},
+		},
+	)
+	s.getExperimentByIDAndCompare(
+		namespace2Code,
+		experiment2ID,
+		&response.GetExperimentResponse{
+			Experiment: &response.ExperimentPartialResponse{
+				ID:   experiment2ID,
+				Name: "UpdatedExperiment2",
+				Tags: []response.ExperimentTagPartialResponse{
+					{
+						Key:   "KeyTag2",
+						Value: "ValueTag2",
+					},
+				},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageActive),
+			},
+		},
+	)
+}
+
+func (s *ExperimentFlowTestSuite) createExperiment(
```

Comment:
this seems like one that should belong to `BaseTestSuite`

Reply:
yes. I mentioned that when I start the task that my goal right now to just cover everything and then we can a bit improve and remove all the duplicates.